astropy/io/misc/ecsv.py

```diff
@@ -409,7 +410,10 @@ def get_data_kwargs(
 
         kw["na_values"] = null_values
         kw["keep_default_na"] = False
-        kw["skiprows"] = header.n_header
+        kw["comment"] = "#"
```

Comment:
this dict could easily be written a a litteral; but that would also create churn and make this harder to review. Could we just move the initialization (`kw = {}`) to line 410 to make this more obvious ?